ddtrace/tracer.py

```diff
@@ -405,49 +427,41 @@ def start_span(
             root_span = tracer.start_span('web.request')
             span = tracer.start_span('web.decoder', child_of=root_span)
 
-        Or if you have a ``Context`` object::
-
-            context = tracer.get_call_context()
-            span = tracer.start_span('web.worker', child_of=context)
+        Be sure to finish all spans to avoid memory leaks and incorrect
+        parenting of spans.
         """
         is_new_process = self._check_new_process()
         if is_new_process:
             # The spans remaining in the context can not and will not be finished
             # in this new process. So we need to copy out the trace metadata needed
             # to continue the trace.
-            if isinstance(child_of, Context):
-                new_ctx = Context(
-                    sampling_priority=child_of._sampling_priority,
-                    span_id=child_of._parent_span_id,
-                    trace_id=child_of._parent_trace_id,
-                )
-                self.context_provider.activate(new_ctx)
-                child_of = new_ctx
-            elif isinstance(child_of, Span):
+            if isinstance(child_of, Span):
+                active = self.context_provider.active()
```

Comment:
Could you explain why it's necessary here to check the context_provider's active?

Reply:
I'll add a comment for this!

Reply:
Addressed in ce0f71f